File: 2015/AI/AIClient_Python/src/event/MissileHitEvent.py
'''
Created on Jan 3, 2015

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World

class MissileHitEvent(IngoingEvent):

    def __init__(self):
        self.hitEntity = 0
        self.hitTeamId = 0
        self.hitCharacterId = 0
        self.originTeamId = 0
        self.originCharacterId = 0
        self.backfire = 0
    
    def fillArguments(self, string):
        arguments = string.split(":", 6)
        
        self.hitEntity = int(arguments[0])
        self.hitTeamId = int(arguments[1])
        self.hitCharacterId = int(arguments[2])
        self.originTeamId = int(arguments[3])
        self.originCharacterId = int(arguments[4])
        self.backfire = int(arguments[5])
        
    def execute(self):
        world = Singleton(World)
        world.getTeam(self.originTeamId).characters[self.originCharacterId]._missileHit()
        if(self.hitEntity == 1):
            world.getTeam(self.hitTeamId).characters[self.hitCharacterId]._hitByMissile()
        elif(self.hitEntity == 2):
            world.getTeam(self.hitTeamId).characters[self.hitCharacterId]._mineHit()
        elif(self.hitEntity == 3):
            world.getTeam(self.hitTeamId).characters[self.hitCharacterId]._missileHit()
        
        if(self.backfire == 1):
            world.getTeam(self.originTeamId).characters[self.originCharacterId]._hitByMissile()

File: 2015/AI/AIClient_JAVA/src/event/MissileHitEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** MissileHitEvent.java
 ** Event that happen when a missile hit an entity
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import world.World;

public class MissileHitEvent extends IngoingEvent {

	private int hitEntity;
	private int hitTeamId;
	private int hitCharacterId;
	private int originTeamId;
	private int originCharacterId;
	private int backfire;

	@Override
	public void execute() {
		World.getInstance().getTeam(originTeamId)
				.getCharacter(originCharacterId).missileHit();

		if (hitEntity == 1) {
			World.getInstance().getTeam(hitTeamId).getCharacter(hitCharacterId)
					.hitByMissile();
		} else if (hitEntity == 2) {
			World.getInstance().getTeam(hitTeamId).getCharacter(hitCharacterId)
					.mineHit();
		} else if (hitEntity == 3) {
			World.getInstance().getTeam(hitTeamId).getCharacter(hitCharacterId)
					.missileHit();
		}
		
		if(backfire == 1)
		{
			World.getInstance().getTeam(originTeamId)
			.getCharacter(originCharacterId).hitByMissile();
		}
	}

	@Override
	public boolean fillArguments(String message) {
		String[] messageParts = message.split(SEPARATOR, 7);

		hitEntity = convertCharToNumeral(messageParts[0]);
		hitTeamId = convertCharToNumeral(messageParts[1]);
		hitCharacterId = convertCharToNumeral(messageParts[2]);

		originTeamId = convertCharToNumeral(messageParts[3]);
		if (originTeamId == 0) {
			return false;
		}
		originCharacterId = convertCharToNumeral(messageParts[4]);
		backfire = convertCharToNumeral(messageParts[5]);
		return true;
	}

}